src/jqLite.js

```diff
@@ -671,7 +671,10 @@ forEach({
       }
       return element.value;
     }
-    element.value = value;
+    // Workaround for Firefox validation #12102.
+    if (element.value != value) {
```

Comment:
Is the `!=` intentional ? It can lead to bugs in corner cases.
E.g. when `value === ''` and one calls `elem.val(0)`, they would expect `elem.val() === '0'`, but since `0 == ''`, nothing will happen (I think).